tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -99,30 +109,43 @@ def test_upgrade_downgrade_brokers(self, from_version, to_version):
         self.zk.start()
 
         # number of nodes needs to be >= 3 for the smoke test
-        self.kafka = KafkaService(self.test_context, num_nodes=3,
+        self.kafka = KafkaService(self.test_context, num_nodes=self.num_kafka_nodes,
                                   zk=self.zk, version=KafkaVersion(from_version), topics=self.topics)
         self.kafka.start()
 
         # allow some time for topics to be created
-        time.sleep(10)
+        wait_until(lambda: self.get_topics_count() >= (len(self.topics) * self.num_kafka_nodes),
+                   timeout_sec=60,
+                   err_msg="Broker did not create all topics in 60 seconds ")
 
         self.driver = StreamsSmokeTestDriverService(self.test_context, self.kafka)
+        self.driver.disable_auto_terminate()
+
         self.processor1 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
         
         self.driver.start()
         self.processor1.start()
-        time.sleep(15)
 
-        self.perform_broker_upgrade(to_version)
+        self.verify_process_records()
+
+        connected_message = "Discovered group coordinator"
+        with self.processor1.node.account.monitor_log(self.processor1.LOG_FILE) as log_monitor:
+            with self.processor1.node.account.monitor_log(self.processor1.STDOUT_FILE) as stdout_monitor:
+
+                self.perform_broker_upgrade(to_version)
+
+                log_monitor.wait_until(connected_message,
+                                       timeout_sec=120,
+                                       err_msg=("Never saw output '%s' on " % connected_message) + str(self.processor1.node.account))
+
+                stdout_monitor.wait_until(self.processed_msg,
+                                          timeout_sec=60,
+                                          err_msg="Never saw output '%s' on" % self.processed_msg + str(self.processor1.node.account))
 
-        time.sleep(15)
-        self.driver.wait()
         self.driver.stop()
 
         self.processor1.stop()
 
-        node = self.driver.node
-        node.account.ssh("grep -E 'ALL-RECORDS-DELIVERED|PROCESSED-MORE-THAN-GENERATED' %s" % self.driver.STDOUT_FILE, allow_fail=False)
```

Comment:
Why remove this line?

Reply:
It's an oversight on my part, I'll add it back.